subt/tools/analyze_detectors.py

```diff
@@ -0,0 +1,243 @@
+                img = cv2.imdecode(np.frombuffer(buf_color, dtype=np.uint8), 1)
+
+                result_cv = cv_detector(img)
+                result_torch = detector(img)
+                result = check_results(result_torch, result_cv)
+                #print(result)
+
+                im_file_name = prefix + "_im_%06d.jpg" % ii
+                im_path = os.path.join(data_dir, im_file_name)
+                for res in result:
+                    print(res)
+                    artf_name, res_t, res_cv = res
+                    score_torch, points = proces_torch_result(res_t)
+                    __, score_cv, bbox = res_cv
+
+                    detection_log.write("%s;%s;%f;%f;%s;%s;%s\r\n" %(str(timestamp), artf_name, score_torch, score_cv,
+                                                                     str(points), str(bbox), im_file_name))
+                    detection_log.flush()
+                    cv2.imwrite(im_path, img)
+
+                ii += 1
+
+    detection_log.close()
+
+
+def plot_data(data):
+    for artf_name in g_artf_names:
+        true_score_t = []
+        true_score_cv = []
+        false_score_t = []
+        false_score_cv = []
+        for artf, score_t, score_cv, detection_type in data:
+            if detection_type == "true" and artf_name == artf:
+                true_score_t.append(score_t)
+                true_score_cv.append(score_cv)
+            if detection_type == "false" and artf_name == artf:
+                false_score_t.append(score_t)
+                false_score_cv.append(score_cv)
+
+        fig = plt.figure(figsize=(7, 6))
+        fig.subplots_adjust(left=0.1, right=0.8, bottom=0.1)
+
+        ax1 = fig.add_subplot(111)
+        ax1.plot(true_score_t, true_score_cv, "go", label="True")
+        ax1.plot(false_score_t, false_score_cv, "ro", label="False")
+        ax1.grid()
+        ax1.set_title(artf_name)
+        ax1.set_xlabel("mdnet (-)")
+        ax1.set_ylabel("cv detector (-)")
+        ax1.legend(bbox_to_anchor=(1.05, 0.7))
+
+        graph_name = artf_name + "_graph"
+        plt.savefig(graph_name, dpi=500)
+        plt.show()
+
+
+def filter_detections(path):
+    # collect data with detection
+    data_to_filter = []
+    dir_list = sorted(os.listdir(path))
+    for dir in dir_list:
+        if not dir.endswith(".log.d"):
+            continue
+        for line in open(os.path.join(path, dir, "detection.log")):
+            __, artf_name, score_t, score_cv, points, bbox, im_name = line.split(";")
+            score_t = float(score_t)
+            score_cv = float(score_cv)
+            points = eval(points)
+            bbox = eval(bbox)
+            im_path = os.path.join(path, dir, im_name)
+            data_to_filter.append([artf_name, score_t, score_cv, points, bbox, im_path, "true"])
+
+    data = manual_sorting(data_to_filter)
+
+    # save data for future analysis
+    with open(os.path.join(path, "detection_overview.csv"), "w") as overview_log:
```

Comment:
https://docs.python.org/3/library/csv.html
Manual csv creation and reading tends to end in tears :-/

Reply:
Ok, time to improve my habits.